src/main/java/com/spotify/docker/client/messages/AuthRegistryConfig.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.docker.client.messages;
+
+import com.google.common.base.Objects;
+
+public class AuthRegistryConfig {
+  private final String repository;
+  private final String username;
+  private final String password;
+  private final String auth;
+  private final String email;
+  private final String serverAddress;
+
+  /**
+   * Wrapper to support X-Registry-Config header.
+   * @param repository  Repository name (uniquely identifies the config)
+   * @param username    User name
+   * @param password    Password for authentication
+   * @param auth      Not used but must be supplied
+   * @param email     EMAIL address of authenticated user
+   * @param serverAddress The address of the repository
+  */
+  public AuthRegistryConfig(String repository, 
+                            String username, 
+                            String password, 
+                            String auth, 
+                            String email, 
+                            String serverAddress) {
+    this.repository = repository;
+    this.username = username;
+    this.password = password;
+    this.auth = auth;
+    this.email = email;
+    this.serverAddress = serverAddress;
+  }
+  
+  /**
+   * A formatted string passed in X-Registry-Config request header.
+   * @return String
+   * {"configs":{"myrepository":{
+   *   "username":"myusername",
+   *   "password":"mypassword",
+   *   "auth":"",
+   *   "email":"myemail",
+   *   "serveraddress":"myserverAddress"}}}
+  */
+  public String toConfig() {
+    StringBuilder sb = new StringBuilder();
```

Comment:
We should use the appropriate jackson stuff rather than building JSON strings by concatenation. The latter could end poorly depending on the values supplied by the user.